### PR TITLE
Deprecate constructor and replace it by another one.

### PR DESCRIPTION
**Pull request by bangerth**
Also update the documentation in a number of places.

### CODE REVIEW

**Comment by bangerth** [comment]
As found while looking over code with @ian-r-rose .

File: doc/news/changes.h [hunk 1]
```diff
@@ -139,6 +139,13 @@ inconvenience this causes.
 <h3>General</h3>
 
 <ol>
+ <li> Changed: As MappingQEulerian before, MappingQ1Eulerian has gained
+ a second constructor that reverts the order of the arguments to indicate
```
Comment (drwells) [inline doc/news/changes.h]: Two not very important style comments:

1. the first clause is is a fragment (should be 'like MappingQEulerian', or 'Similiar to MappingQEulerian', or similar).
2. 'reverses' or 'switches', not 'reverts'

**Comment by drwells** [comment]
I can't seem to click through to it, but there is a code example in the class documentation (lines 67 through 73) that uses the old constructor. Could you also update that?

Line 82 references `EulerVectorType` as the name of a template parameter: this should be `VectorType`. It is not related to the current PR, but you did fix some other small things in the  documentation.

**Comment by drwells** [comment]
Clarification: the header file consistently uses `VectorType`, not `EulerVectorType`, as the template parameter, so the documentation is a bit misleading.

It looks like the source file uses `EulerVectorType`. I'll do a search and replace after this is merged.

**Comment by bangerth** [comment]
All good comments. I've updated the pull request and also removed a comment in both classes that pertained to the copy constructor of `Vector` not being explicit, and that is no longer correct. It was also really difficult to understand if you didn't know the issue...

**Comment by drwells** [comment]
Nice catch with the explicit thing.

**Pull request merged**